src/lib.rs

```diff
@@ -446,7 +471,7 @@ mod arith_impl {
     pub fn mac_with_carry(a: u64, b: u64, c: u64, carry: &mut u64) -> u64 {
         use std::num::Wrapping;
 
-        let tmp = (u128::from(a)).wrapping_add(u128::from(b).wrapping_mul(u128::from(c))).wrapping_add(u128::from(*carry));
+        let tmp = (u128::from(a)).wrapping_add(mul_u64(b, c)).wrapping_add(u128::from(*carry));
```

Comment:
Does this `mul_u64` acts the same as previously used `wrapping_mul` in this context?

Reply:
I believe so. The product of two u64 numbers shouldn't overflow in any case.